### ISSUE TITLE
Use distroless base image for images

### ISSUE THREAD

**Issue by aramase** [Azure/aad-pod-identity#595]
- Use [distroless](https://github.com/GoogleContainerTools/distroless) base image for pod-identity components

**Comment by elsesiy** [comment]
@aramase I've been using distroless extensively but from a cluster operator perspective it comes with the huge downside of not being able to exec into the pod for troubleshooting purposes. What's the reasoning behind switching? Because it's a privileged pod with access to iptables?

**Comment by aramase** [comment]
@elsesiy Using `static` as base image we could avoid os vulnerabilities. That's the idea behind switching to use those. This can be done for MIC, but I still don't think we can do it for NMI because it requires iptables.

**Comment by elsesiy** [comment]
@aramase Yeah I understand the security implications, just wanted to point out the operator implications as well 👍
